### DISCUSSION TITLE
Register and enroll users removal

### DISCUSSION THREAD

**Discussion by rajat-dlt** [hyperledger/fabric-sdk-node#673] [Q&A] [answered]
I was planning to give a fix for an issue I found in the `fabric-ca-client`, the grpc connections were not getting reused. If we're using a higher Node Version.

But the problem is fabric-ca-client states that it is deprecated with Fabric V2.5. 

And in the [fabirc-gateway](https://hyperledger.github.io/fabric-gateway/), there's no option to register and enroll users. Even the samples use the created certificates. Using the fabric-gateway, how is a user supposed to register and enroll new users ? 

This is very confusing for a new person as he won't understand how to register users on the network.

**Comment by rajat-dlt** [comment]
@bestbeforetoday, maybe I'm missing something. Could you please shed light on this?

**Comment by bestbeforetoday** [comment, marked as answer]
As I understand it, the direction (since Fabric v2.2 was released) was for the Fabric CLI commands to be the primary tool for performing administrative operations. The overhead of maintaining the (already inconsistent) admin capability implemented in each of the language SDKs, in addition to the CLI commands, made it difficult to continue that approach. 

There has been some effort made to provide a specific admin-focused API in the [hyperledger/fabric-admin-sdk](https://github.com/hyperledger/fabric-admin-sdk) repository, but the focus has been on a Go implementation (to support Kubernetes/Cloud deployment). This only covers channel creation/configuration and chaincode deployment, not registering and enrolling users with the Fabric CA since - at least in theory - the Fabric CA is an optional rather than core component and can be replaced by an alternative CA.

Having said all that, the fabric-ca-client in the Node SDK is not disappearing and should still continue to work with the Fabric CA. I am happy to accept community contributions to fix issues or even (within reason) add enhancements.

**Comment by rajat-dlt** [comment]
Thanks for such a detailed response. I'll go ahead and raise a PR for the fix that I think might really help.

**Comment by rajat-dlt** [comment]
@bestbeforetoday, 
I've added a PR: #674 
Let me know if this seems fine or your views on this. This small fix has helped my team immensely in case of high throughput on the fabric-ca.

**Comment by bh4rtp** [comment]
what is the relation of `fabric-gateway` between `fabric-node-sdk` and `fabric-gateway-java`? I remember hyperledger also has a repo `fabric-gateway-go` and cannot find it now. Shall we use `fabric-gateway` instead of `fabric-node-sdk` and `fabric-gateway-java` for fabric release 2.4+?

**Reply by bestbeforetoday** [reply]
The top of the README for both this repository and the fabric-gateway-java repository say:

> Note: This API is deprecated as of Fabric v2.5. When developing applications for Hyperledger Fabric v2.4 and later, you should use the [Fabric Gateway client API](https://hyperledger.github.io/fabric-gateway/).
